plugins/analysis/server/plugin.go

```diff
@@ -105,97 +91,32 @@ func getMaxPacketSize(packetSizes ...int) int {
 	return maxPacketSize
 }
 
-func processIncomingPacket(connectionState *byte, receiveBuffer *[]byte, conn *network.ManagedConnection, data []byte, offset *int, connectedNodeId *string) {
-	firstPackage := *connectionState == STATE_INITIAL
-
-	if firstPackage || *connectionState == STATE_CONSECUTIVE {
-		var err error
-		if *connectionState, *receiveBuffer, err = parsePackageHeader(data); err != nil {
-			Events.Error.Trigger(err)
-
-			conn.Close()
-
-			return
-		}
-
-		*offset = 0
+func processIncomingPacket(connectionState *byte, receiveBuffer *[]byte, conn *network.ManagedConnection, data []byte) {
+	var err error
+	if *connectionState, *receiveBuffer, err = parsePackageHeader(data); err != nil {
+		Events.Error.Trigger(err)
 
-		switch *connectionState {
-		case STATE_ADD_NODE:
-			*receiveBuffer = make([]byte, addnode.MARSHALED_TOTAL_SIZE)
-
-		case STATE_PING:
-			*receiveBuffer = make([]byte, ping.MARSHALED_TOTAL_SIZE)
-
-		case STATE_CONNECT_NODES:
-			*receiveBuffer = make([]byte, connectnodes.MARSHALED_TOTAL_SIZE)
-
-		case STATE_DISCONNECT_NODES:
-			*receiveBuffer = make([]byte, disconnectnodes.MARSHALED_TOTAL_SIZE)
-
-		case STATE_REMOVE_NODE:
-			*receiveBuffer = make([]byte, removenode.MARSHALED_TOTAL_SIZE)
-		}
-	}
+		conn.Close()
```

Comment:
This is very strange, but I guess it is the way how the server is designed. Usually you would do a `defer conn.Close()` right after the open to assure that it is always closed, but I am not sure this is applicable here.